### ISSUE TITLE
Vanilla Renderers question

### ISSUE THREAD

**Issue by leifoolsen** [eclipsesource/jsonforms#1003]
Hello. I have just started exploring JsonForms, have a few questions.

1. Since we have an established design system with PostCSS and CSS modules, it seems that vanilla rendrers is the right package to use. Where can I find documentation of available CSS classes for vanilla renderers, in particular, I am interested in the documentation for CSS classes for components that render layout.

2. With the vanilla package, it seems that the class names for vertical and horizontal layout are empty; see the attached screenshot. Is that the intention? If I switch to MaterialUI renderers, classes for vertical and horizontal layout are not empty. How do I style horizontal and vertical layout with vanilla? 

Regards
Leif Olsen

**Comment by leifoolsen** [comment]
![skjermbilde-17](https://user-images.githubusercontent.com/10088486/41354125-70b20fec-6f1e-11e8-85d5-f55a50899eb8.PNG)

**Comment by edgarmueller** [comment]
Hi, thanks for giving JSONForms a try!
The vanilla renderer set is based on a custom `styling` reducer which allows to associate class names with predefined ids/styles. Renderers are then able to query the styling substate via `getStyle` and `getStyleAsClassName` in order to retrieve the respective class names to be applied.
Currently, there is no documentation what these predefined style constants are, but we'll try to update the documentation soon. 
Until then you can browse the source and look for callers of `getStyle` and `getStyleAsClassName`, e.g. in the case of the HorizontalLayout:

```js
const layoutClassName = getStyleAsClassName('horizontal.layout');
const childClassNames = getStyle('horizontal.layout.item', elementsSize)
    .concat(['horizontal-layout-item'])
    .join(' ');

  return (
    <JsonFormsLayout
      className={layoutClassName}
      visible={visible}
    >
      {renderChildren(horizontalLayout, schema, childClassNames, path)}
    </JsonFormsLayout>
  );
```
So if there are no class names for 'horizontal.layout' registered, the class names stay empty. 
Hope this helps, we are happy about any feedback.

**Comment by leifoolsen** [comment]
@edgarmueller Thank youfor a quick reply.

I had the impression that these classes were already registered since I could see some of the class names when inspecting the browser output. Are some of these classes hard-coded?

I can find class names in `vanilla/src/util/index.ts`, but I think they are not exported (public).  Can you give an example of how I register the styles provided in `vanilla/src/util/index.ts` (or a similar struct with class names)?

When I have this:

```javascript
const store = createStore(
  combineReducers({ jsonforms: jsonformsReducer() }),
  {
    jsonforms: {
      fields: vanillaFields,
      renderers: vanillaRenderers,
    },
  }
);
```
Where do I register the CSS classes?

**Comment by edgarmueller** [comment]
@leifoolsen Sorry for the late answer. Yes, you are right, we'll add those to the vanilla/src/index.ts. I'll let you know once this is done and the documentation has been updated.

**Comment by edgarmueller** [comment]
We've released 2.0.2, update of seeds and docs will follow the next couple of days. Until then, we've added a markdown file which lists all available vanilla styles constants, see https://github.com/eclipsesource/jsonforms/blob/master/packages/vanilla/Styles.md.

**Comment by edgarmueller** [comment]
We've added a new section on [Renderer sets](https://jsonforms.io/docs/renderer-sets) that contains most of the info we referred to in this issue. We therefore close this one now. If you feel like anythings is missing let us know.
